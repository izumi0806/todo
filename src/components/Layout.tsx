import type { ReactNode } from "react";

type LayoutProps = {
	children: ReactNode;
};

const Layout = ({ children }: LayoutProps) => {
	return (
		<div className="flex min-h-screen justify-center bg-gray-100">
			<div className="w-10/12 min-h-[90%] bg-white shadow-lg rounded-xl p-6 my-8">
				{children}
			</div>
		</div>
	);
};

export default Layout;
