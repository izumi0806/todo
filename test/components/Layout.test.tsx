import { render } from "@testing-library/react";
import React from "react";
import Layout from "../../src/components/Layout";

describe("Layout", () => {
	it("should render", () => {
		const { container } = render(
			<Layout>
				<div>Test Content</div>
			</Layout>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
