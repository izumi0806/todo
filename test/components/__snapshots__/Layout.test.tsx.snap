// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > should render 1`] = `
<div
  class="flex min-h-screen justify-center bg-gray-100"
>
  <div
    class="w-10/12 min-h-[90%] bg-white shadow-lg rounded-xl p-6 my-8"
  >
    <div>
      Test Content
    </div>
  </div>
</div>
`;
